lib/spack/spack/cmd/mirror.py

```diff
@@ -93,7 +92,15 @@ def setup_parser(subparser):
         '--scope', choices=scopes, metavar=scopes_metavar,
         default=spack.config.default_modify_scope(),
         help="configuration scope to modify")
-
+    add_parser.add_argument(
+        '-s3-id', '--s3-access-key-id',
+        help="ID string to use to connect to this S3 mirror")
+    add_parser.add_argument(
+        '-s3-key', '--s3-access-key-secret',
+        help="Secret string to use to connect to this S3 mirror")
+    add_parser.add_argument(
+        '-s3-token', '--s3-access-token',
+        help="Access Token to use to connect to this S3 mirror")
```

Comment:
I'm not a fan of these short flags that aren't actually short.  Better would be to just have the long options, imo.